internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -129,39 +129,39 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 		if dErr := preUpgradeCleanup(u.agentInfo.Version()); dErr != nil {
 			u.log.Errorf("Unable to remove file after verification failure: %v", dErr)
 		}
-		return nil, err
+		return nil, errors.New(err, errors.TypeRetryableAction)
 	}
 
 	newHash, err := u.unpack(ctx, version, archivePath)
 	if err != nil {
-		return nil, err
+		return nil, errors.New(err, errors.TypeRetryableAction)
 	}
 
 	if newHash == "" {
-		return nil, errors.New("unknown hash")
+		return nil, errors.New("unknown hash", errors.TypeRetryableAction)
 	}
 
 	if strings.HasPrefix(release.Commit(), newHash) {
 		return nil, ErrSameVersion
 	}
 
 	if err := copyActionStore(newHash); err != nil {
-		return nil, errors.New(err, "failed to copy action store")
+		return nil, errors.New(err, "failed to copy action store", errors.TypeRetryableAction)
 	}
 
 	if err := ChangeSymlink(ctx, newHash); err != nil {
 		rollbackInstall(ctx, newHash)
-		return nil, err
+		return nil, errors.New(err, errors.TypeRetryableAction)
 	}
 
 	if err := u.markUpgrade(ctx, newHash, action); err != nil {
 		rollbackInstall(ctx, newHash)
-		return nil, err
+		return nil, errors.New(err, errors.TypeRetryableAction)
 	}
 
 	if err := InvokeWatcher(u.log); err != nil {
 		rollbackInstall(ctx, newHash)
-		return nil, errors.New("failed to invoke rollback watcher", err)
+		return nil, errors.New("failed to invoke rollback watcher", err, errors.TypeRetryableAction)
```

Comment:
This doesn't seem like something we need to place everywhere. Because as soon as a developer forgets to add that error they no longer get retried. I would just say if an action errors and its retryable we retry it.

Reply:
Do we want to consider all errors retryable? We know that `ErrSameVersion` will never succeed.